e2e/ctl_v3_migrate_test.go

```diff
@@ -0,0 +1,149 @@
+// Copyright 2016 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"fmt"
+	"os"
+	"sort"
+	"testing"
+	"time"
+
+	"github.com/coreos/etcd/client"
+	"github.com/coreos/etcd/etcdserver/etcdserverpb"
+	"github.com/coreos/etcd/pkg/testutil"
+)
+
+func TestCtlV3Migrate(t *testing.T) {
+	defer testutil.AfterTest(t)
+
+	// 1. start 3-node cluster for v2
+	epc := setupEtcdctlTest(t, &configNoTLS, true)
+	defer func() {
+		if errC := epc.Close(); errC != nil {
+			t.Fatalf("error closing etcd processes (%v)", errC)
+		}
+	}()
+
+	keys := make([]string, 3)
+	vals := make([]string, 3)
+	for i := range keys {
+		keys[i] = fmt.Sprintf("foo_%d", i)
+		vals[i] = fmt.Sprintf("bar_%d", i)
+	}
+
+	// 2. write keys to v2 store
+	var v2Resps v2ResponseByCreation
```

Comment:
i think the get-json and sort-response thing are over kill. We know what we put into v2. We just need to compare what we get from v3 after migration against what we put into v2. We do not need to compare it with the responses we get out of v2.